software/contrib/beat_writer.py

```diff
@@ -0,0 +1,165 @@
+from time import sleep_ms, ticks_ms, ticks_diff
+from europi import *
+from europi_script import EuroPiScript
+from random import randint
+from machine import freq
+
+freq(250_000_000)
+
+class BeatWriter(EuroPiScript):
+    def __init__(self):
+        super().__init__()
+        
+        self.length = 16
+        
+        state = self.load_state_json()
+        self.rhythm_1 = state.get("rhythm_1", self.create_blank_rhythm(self.length))
+        self.rhythm_2 = state.get("rhythm_2", self.create_blank_rhythm(self.length))
+        
+        self.previous_step_ticks = ticks_ms
+        self.previous_step_length = 0
+    
+        self.step = 0
+                
+        self.din_out = cv1
```

Comment:
[optional] For consistency with the fields below:

```suggestion
        self.din_output = cv1
```